internal/datastore/common/gc.go

```diff
@@ -97,7 +100,13 @@ func StartGarbageCollector(ctx context.Context, gc GarbageCollector, interval, w
 
 		case <-time.After(interval):
 			err := collect(gc, window, timeout)
-			if err != nil {
+			switch {
+			case err == nil:
+				break
+			case errors.As(err, &datastore.ErrReadOnly{}):
```

Comment:
Shouldn't this shutdown the GC worker?

Reply:
Yeah, I wasn't sure if we should shutdown the worker or just continue to no-op.
I figured running it like this means that the lifecycle/cleanup stays consistent regardless of read-only.

Reply:
It does, but at the same time it means a lot of extra debug logs... so I think we should just shut down the process entirely

Reply:
I could also just make it trace() instead of debug